src/aspire/estimation/covar2d.py

```diff
@@ -267,10 +267,17 @@ def get_cwf_coeffs(self, coeffs, ctf_fb, ctf_idx, mean_coeff=None, covar_coeff=N
         if covar_coeff is None:
             covar_coeff = self.get_covar(coeffs, ctf_fb, ctf_idx, mean_coeff, noise_var=noise_var)
 
-        blk_partition = blk_diag_partition(ctf_fb[0])
-
+        blk_partition = blk_diag_partition(covar_coeff)
         noise_covar_coeff = blk_diag_mult(noise_var, blk_diag_eye(blk_partition, dtype=self.as_type))
 
+        if (ctf_fb is None) or (ctf_idx is None):
```

Comment:
Why not just use the `ctf_fb = blk_diag_eye` trick above? That way it's easier to see what's going on.

Reply:
As above. This tries to set the case of clean images of simulation data. But I think it probably not necessary and should be removed.

Reply:
I have merged the case for simulated clean images and add more notation in commit 059f18f.

Reply:
As above. Changed to use the identity filter when the CTF filters are not specified in commit b6a96b. The `coeffs` and `noise_var` will determine whether the images are clean or not.